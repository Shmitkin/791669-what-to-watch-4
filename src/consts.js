export const MOVIE_RATING_TITLES = [
  {rating: 10, title: `Awesome`},
  {rating: 8, title: `Very good`},
  {rating: 5, title: `Good`},
  {rating: 3, title: `Normal`},
  {rating: 0, title: `Bad`},
];

export const MovieInfoTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const DEFAULT_GENRE = `All genres`;
export const NAV_MAX_GENRES = 10;

export const Movies = {
  SHOW_MORE_COUNT: 8,
  DEFAULT_SHOW_COUNT: 8
};

export const MAX_SIMILAR_MOVIES = 4;
