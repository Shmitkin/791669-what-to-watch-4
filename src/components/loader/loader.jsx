import PropTypes from "prop-types";
import {connect} from "react-redux";

import {getDataLoadStatus} from "../../reducer/data/selectors.js";
import {DataLoadStatus} from "../../consts.js";


function Loader({children, requiredData, isDataLoaded}) {
  const loadStatus = requiredData.map((data) => isDataLoaded(data));

  return loadStatus.indexOf(false) === -1
    ? children
    : null;
}

Loader.propTypes = {
  isDataLoaded: PropTypes.func.isRequired,
  requiredData: PropTypes.arrayOf(
      PropTypes.oneOf(Object.values(DataLoadStatus))
  ).isRequired,
  children: PropTypes.object.isRequired,
};

const MapStateToProps = (state) => ({
  isDataLoaded: (data) => getDataLoadStatus(state, data),
});

export {Loader};
export default connect(MapStateToProps)(Loader);
