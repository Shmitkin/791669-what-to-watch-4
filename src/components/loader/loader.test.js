import React from "react";
import renderer from "react-test-renderer";

import {Loader} from "./loader.jsx";
import {DataLoadStatus} from "../../consts.js";

it(`Should Loader render correctly`, () => {

  const tree = renderer
    .create(
        <Loader isDataLoaded={()=>{
          return true;
        }}
        requiredData={[DataLoadStatus.MOVIES, DataLoadStatus.PROMO_MOVIE]}>

          <div>Test</div>

        </Loader>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
