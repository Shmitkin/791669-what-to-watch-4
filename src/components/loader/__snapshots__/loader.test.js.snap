// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Loader render correctly 1`] = `
<div>
  Test
</div>
`;
